sources/Renderer/RenderContext.cpp

```diff
@@ -86,44 +86,13 @@ bool RenderContext::SetVsync(const VsyncDescriptor& vsyncDesc)
 
 void RenderContext::SetOrCreateSurface(const std::shared_ptr<Surface>& surface, VideoModeDescriptor videoModeDesc, const void* windowContext)
 {
-    if (surface)
-    {
-        /* Get and output resolution from specified window */
-        videoModeDesc.resolution = surface->GetContentSize();
-        surface_ = surface;
-    }
-    else
-    {
-        #ifdef LLGL_MOBILE_PLATFORM
-
-        /* Create new canvas for this render context */
-        CanvasDescriptor canvasDesc;
-        {
-            canvasDesc.borderless = videoModeDesc.fullscreen;
-        }
-        surface_ = Canvas::Create(canvasDesc);
-
-        #else
-
-        /* Create new window for this render context */
-        WindowDescriptor windowDesc;
-        {
-            windowDesc.size             = videoModeDesc.resolution;
-            windowDesc.borderless       = videoModeDesc.fullscreen;
-            windowDesc.centered         = !videoModeDesc.fullscreen;
-            windowDesc.windowContext    = windowContext;
-        }
-        surface_ = Window::Create(windowDesc);
-
-        #endif
-    }
-
-    /* Store video mode settings */
+    // XXX TODO
```

Comment:
I would like to find another solution that still provides a default surface. Are you referring to the information for resolution scaling or the preferred resolution? I think that should be doable.

Reply:
By default surface you mean a Window which strikes me as rather focused on games which always fill the full window (shades of SDL). I would like to generalize LLGL so it can be used in other context e.g. multiple NSViews, individual CALayer's. In the app I'm working on I don't even have a need for the Window surface at all. But I'll think about it.

Reply:
The `Window` class is not necessary for an LLGL application as it merely adds a few more features like positioning and keyboard/mouse event handling so you can set up a prototype fairly easy. As soon as you want a more complex application you would ideally write your own implementation that derives from `Surface` as this class provides the minimum required interface for all backends. "View" is a reasonable term as well but more common in the macOS/iOS space. I think there are several parts in this PR we should split up for a separate discussion and approach.